app/electron/main.ts

```diff
@@ -453,6 +453,42 @@ class PluginManagerEventListeners {
   }
 }
 
+function getShellEnv(): NodeJS.ProcessEnv {
+  if (process.platform === 'win32') {
+    return process.env;
+  }
+
+  let shell = process.env.SHELL || '/bin/sh';
+  if (process.platform === 'darwin') {
+    try {
+      shell = userInfo().shell || '/bin/zsh';
+    } catch (error) {
+      shell = '/bin/zsh';
+    }
+  }
+
+  try {
+    const env = { ...process.env, DISABLE_AUTO_UPDATE: 'true' };
+    const result = execSync(`${shell} -ilc 'env'`, {
```

Comment:
Please do not use sync function here. We need to not block the main thread.

Reply:
Even if we change getShellEnv to be asynchronous, the startServer function that calls it is still synchronous. This means the main thread will block until getShellEnv completes, regardless of whether it's sync or async.

Reply:
Not so.

An async function doesn't block if it's called in a sync function. Additionally, higher level functions can be changed.